itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java

```diff
@@ -1430,82 +1438,92 @@ public Boolean apply(@Nullable NotificationEvent entry) {
             .run(" drop database if exists " + replicatedDbName_CM + " cascade");
   }
 
-  private void injectableForTaskCreation(String primarydb, String replicadb, String tbl, String eventType,
-                                             String eventTypeExpect, WarehouseInstance.Tuple tuple) throws Throwable {
-    List<String> withConfigs = Arrays.asList("'hive.repl.move.optimized.scheme'= ' hDfs , '");
-
-    // fail add notification for given event type.
-    BehaviourInjection<NotificationEvent, Boolean> callerVerifier
-            = new BehaviourInjection<NotificationEvent, Boolean>() {
-      @Nullable
-      @Override
-      public Boolean apply(@Nullable NotificationEvent entry) {
-        if (entry.getEventType().equalsIgnoreCase(eventType) && entry.getTableName().equalsIgnoreCase(tbl)) {
-          injectionPathCalled = true;
-          LOG.warn("Verifier - DB: " + String.valueOf(entry.getDbName())
-                  + " Table: " + String.valueOf(entry.getTableName())
-                  + " Event: " + String.valueOf(entry.getEventType()));
-          return false;
-        }
-        if (entry.getEventType().equalsIgnoreCase(eventTypeExpect) && entry.getTableName().equalsIgnoreCase(tbl)) {
-          nonInjectedPathCalled = true;
-        }
+  private abstract class checkTaskPresent {
+    public boolean hasTask(Task rootTask) {
+      if (validate(rootTask)) {
         return true;
       }
+      List<Task<? extends Serializable>> childTasks = rootTask.getChildTasks();
+      if (childTasks == null) {
+        return false;
+      }
+      for (Task<? extends Serializable> childTask : childTasks) {
+        if (hasTask(childTask)) {
+          return true;
+        }
+      }
+      return false;
+    }
+
+    public abstract boolean validate(Task task);
+  }
+
+  private boolean hasMoveTask(Task rootTask) {
+    checkTaskPresent validator =  new checkTaskPresent() {
+      public boolean validate(Task task) {
+        return  (task instanceof MoveTask);
+      }
     };
+    return validator.hasTask(rootTask);
+  }
 
-    InjectableBehaviourObjectStore.setAddNotificationModifier(callerVerifier);
-    try {
-      replica.load(replicadb, tuple.dumpLocation, withConfigs);
-    } finally {
-      InjectableBehaviourObjectStore.resetAddNotificationModifier();
-    }
+  private boolean hasPartitionTask(Task rootTask) {
+    checkTaskPresent validator =  new checkTaskPresent() {
+      public boolean validate(Task task) {
+        if (task instanceof DDLTask) {
+          DDLTask ddlTask = (DDLTask)task;
+          if (ddlTask.getWork().getAddPartitionDesc() != null) {
+            return true;
+          }
+        }
+        return false;
+      }
+    };
+    return validator.hasTask(rootTask);
+  }
 
-    callerVerifier.assertInjectionsPerformed(false, true);
+  private void checkTaskCreation(String replicadb, boolean hasMoveTask, boolean hasPartitionTask, boolean isIncDump,
```

Comment:
Instead of flags hasMoveTask and hasPartitionTask, can we use array of enum so that this method can be reusable for future.

Reply:
changed the method to retrun the task. That way in can be used for any purpose. traverse execute etc